Dockerfile

```diff
@@ -0,0 +1,46 @@
+# Prepare
+
+FROM node:8.4.0-alpine AS base
+
+RUN mkdir -p /opt/app
+
+WORKDIR /opt/app
+
+# Install
+
+FROM base AS dependencies
+
+COPY package.json .
+COPY yarn.lock .
+
+RUN yarn --production --ignore-scripts --prefer-offline
+RUN cp -R node_modules production_node_modules
+RUN yarn
+
+# Build
+
+FROM base AS build
```

Comment:
Why are build and dependencies in different stages?

Reply:
The idea was to have an image per logical step: base,install,build,(test?),run
It does not serve any practical purpose at the moment.
Does that bother you?

Reply:
Not sure, do these stages add any performance overhead?

Reply:
A bit, probably. Haven't noticed any negative influence on performance so far nor have seen any mention of serious impact elsewhere. I would like to keep it like that for now, this structure helps me reason about what happens on which stage better.